lib/metasploit/framework/password_crackers/jtr/formatter.rb

```diff
@@ -9,19 +9,29 @@ module Formatter
           # @param cred [credClass] A credential from framework.db
           # @return [String] The hash in jtr format or nil on no match.
           def self.hash_to_jtr(cred)
-            case cred.private.type
-            when 'Metasploit::Credential::NTLMHash'
-              return "#{cred.public.username}:#{cred.id}:#{cred.private.data}:::#{cred.id}"
-            when 'Metasploit::Credential::PostgresMD5'
-              if cred.private.jtr_format =~ /postgres|raw-md5/
+            params_to_jtr(
+              (cred.public.nil? ? '' : cred.public.username),
+              cred.private.data,
+              cred.class.model_name.element.to_sym,
+              format: cred.private.jtr_format,
+              db_id: cred.id
+            )
+          end
+
+          def self.params_to_jtr(username, private_data, private_type, format: nil, db_id: nil)
```

Comment:
I changed this around so Metasploit can get the JtR format of secrets without relying on a database connection. This means the `auxiliary/gather/ldap_passwords` module can use this code to format the hashes regardless of whether or not the user has a database connected.